src/primitives.rs

```diff
@@ -31,6 +31,20 @@ impl fmt::Debug for Bytes32 {
     }
 }
 
+impl Bytes32 {
+    pub fn xor(&self, other: Self) -> Self {
+        let mut result = Vector::default();
+        for (i, (a, b)) in self.0.iter().zip(other.0.iter()).enumerate() {
+            result[i] = a ^ b;
+        }
+        Self(result)
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_slice()
+    }
```

Comment:
i haven't looked more deeply into the specifics but i'd *like* to avoid the need for methods like this, even if it means keeping in the inner type `pub` in the crate... i'm not completely sure we can around it though (e.g. if a user of this crate needs the underlying `&[u8]` some day for a `Bytes32`...